docs/components/motor/28byj-48.md

```diff
@@ -0,0 +1,104 @@
+
+{{< alert title="Tip" color="tip" >}}
+Viam also supports some more advanced stepper driver chips ([TMC5072](../tmc5072/), [DMC4000](../dmc4000/)) that have their own microcontrollers that handle things like speed and acceleration control.
+{{< /alert >}}
+
+## Configuration
+
+To configure a `28byj48` motor as a component of your robot, first configure the [board](/components/board/) to which the motor driver is wired.
+
+Then assign your motor a `name` (to identify the motor), `type` (`motor`) and `model` (`28byj48`), and fill in the [attributes](#required-attributes) that apply to your particular hardware.
+Refer to your motor and motor driver data sheets for specifics.
+
+{{< tabs name="gpiostepper-config">}}
+{{% tab name="Config Builder" %}}
+
+Example configuration for a `28byj48` stepper motor:
+
+![A 28byj48 motor config.](../../img/motor/28byj48-config-ui.png)
+
+{{% /tab %}}
+{{% tab name="JSON Template" %}}
+
+```json
+{
+  "components": [
+    {
+      "name": <board_name>,
+      "type": "board",
+      "model": <board_model>,
+      "attributes": {},
+      "depends_on": [],
+    },
+    {
+      "name": <motor_name>,
+      "type": "motor",
+      "model": "28byj48",
+      "attributes": {
+        "board": <board_name>,
+        "pins": {
+          "in1": <board pin number (string) of in1 pin>,
+          "in2": <board pin number (string) of in2 pin>,
+          "in3": <board pin number (string) of in3 pin>,
+          "in4": <board pin number (string) of in4 pin>
+        },
+        "ticks_per_rotation": <int>
+      },
+      "depends_on": []
+    }
+  ]
+}
+```
+
+{{% /tab %}}
+{{% tab name="JSON Example" %}}
+
+Example configuration for a `28byj48` stepper motor:
+
+```json
+{
+  "components": [
+    {
+      "name": "example-board",
+      "type": "board",
+      "model": "pi"
+    },
+    {
+      "name": "example-motor",
+      "type": "motor",
+      "model": "28byj48",
+      "attributes": {
+        "board": "example-board",
+        "pins": {
+          "in1": "16",
+          "in2": "18",
+          "in3": "29",
+          "in4": "22"
+        },
+        "ticks_per_rotation": 200
+      }
+    }
+  ]
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+### Required Attributes
+
+Name | Type | Description
+-------------- | ---- | ---------------
+`board` | string | Should match name of board to which the motor driver is wired.
+`pins` | object | A json object containing `in1`, `in2`, `in3`, and `in4` pin numbers.
+`ticks_per_rotation` | integer | Number of full steps in a rotation. 200 (equivalent to 1.8 degrees per step) is very common. If your data sheet specifies this in terms of degrees per step, divide 360 by that number to get ticks per rotation.

```

Comment:
Oooh that is possibly a good thing to mention RE what kind of motor and driver this is for? Since that's different from "standard" stepper motors?

Reply:
This is specifically for 28byj48 motor. The motor has a gear ratio of 64:1. which is  how we were able to calculate 4096 steps per revolution. This may be different for other stepper motors depending on what kind of gear ratio they have.